From 6911104496b782f89b0e4ad0967eb877b189e12c Mon Sep 17 00:00:00 2001
From: Hongcheng Xie <hongcheng.xie@intel.com>
Date: Wed, 9 Sep 2020 13:30:12 +0800
Subject: [PATCH] Temp ignore failure in map swap synchronize_rcu

Temp still return ok even netd happen "map swap synchronize_rcu() failed"
error in swapActiveStatsMap() to avoid system_server crash.

Change-Id: Icd1680d1818428b4b7dd0318b03741522b096087
Tracked-On: ACP-1276
Signed-off-by: Hongcheng Xie <hongcheng.xie@intel.com>
---
 server/TrafficController.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/server/TrafficController.cpp b/server/TrafficController.cpp
index 3839962f..8e544c23 100644
--- a/server/TrafficController.cpp
+++ b/server/TrafficController.cpp
@@ -802,7 +802,9 @@ Status TrafficController::swapActiveStatsMap() {
     int ret = synchronizeKernelRCU();
     if (ret) {
         ALOGE("map swap synchronize_rcu() ended with failure: %s", strerror(-ret));
-        return statusFromErrno(-ret, "map swap synchronize_rcu() failed");
+        // Temp return ok even synchronize_rcu() error to avoid framework crash
+        // return statusFromErrno(-ret, "map swap synchronize_rcu() failed");
+        return netdutils::status::ok;
     }
     return netdutils::status::ok;
 }
-- 
2.17.1

